src/server/pfs/server/driver_file.go

```diff
@@ -347,6 +347,12 @@ func (d *driver) walkFile(ctx context.Context, file *pfs.File, cb func(*pfs.File
 	if p == "/" {
 		p = ""
 	}
+	if number == 0 {
```

Comment:
This should have the same requirements as a list file pagination, so we should probably just refactor the check into a utility function (i.e. number == 0 && reverse is an error and number > 100000 is an error).